### PR TITLE
fix(helm): inject loggingRef into FluentbitAgent

### PR DESCRIPTION
**Pull request by aslafy-z**
Inject `.Values.logging.loggingRef` into FluentbitAgent if not given in `.Values.logging.fluentbit.loggingRef`

### CODE REVIEW

File: charts/logging-operator/values.yaml [hunk 1]
```diff
@@ -136,7 +136,7 @@ logging:
   enabled: false
 
   # -- Reference to the logging system. Each of the loggingRefs can manage a fluentbit daemonset and a fluentd statefulset.
-  loggingRef: ""
+  loggingRef: "logging"
```
Comment (pepov) [inline charts/logging-operator/values.yaml:139]: can we leave this empty for the sake of backward compatiblity with the existing tests?

File: charts/logging-operator/templates/logging_fluentbit.yaml [hunk 2]
```diff
@@ -5,7 +5,11 @@ metadata:
   name: {{ include "logging-operator.releasename" . }}
   labels: {{ include "logging-operator.labels" . | nindent 4 }}
 {{- if .Values.logging.fluentbit }}
-spec: {{ toYaml .Values.logging.fluentbit | nindent 2 }}
+spec:
+  {{- with .Values.logging.loggingRef }}
+  loggingRef: {{ . }}
+  {{- end }}
+  {{- toYaml .Values.logging.fluentbit | indent 2 }}
```
Comment (pepov) [inline charts/logging-operator/templates/logging_fluentbit.yaml:12]: shouldn't this be nindent?

File: charts/logging-operator/templates/logging_fluentbit.yaml [hunk 3]
```diff
@@ -1,11 +1,17 @@
 {{ if and .Values.logging.enabled (not .Values.logging.fluentbitDisabled) -}}
+
+{{- $fluentbitSpec := .Values.logging.fluentbit -}}
+{{- if or .Values.logging.loggingRef .Values.logging.fluentbit.loggingRef -}}
+  {{- $fluentbitSpec := set $fluentbitSpec "loggingRef" (default .Values.logging.loggingRef .Values.logging.fluentbit.loggingRef -}}
```
Comment (aslafy-z) [inline charts/logging-operator/templates/logging_fluentbit.yaml:5]:
```suggestion
  {{- $fluentbitSpec := set $fluentbitSpec "loggingRef" (default .Values.logging.loggingRef .Values.logging.fluentbit.loggingRef) -}}
```

File: charts/logging-operator/templates/logging_fluentbit.yaml [hunk 4]
```diff
@@ -1,11 +1,17 @@
 {{ if and .Values.logging.enabled (not .Values.logging.fluentbitDisabled) -}}
+
+{{- $fluentbitSpec := .Values.logging.fluentbit -}}
+{{- if or .Values.logging.loggingRef .Values.logging.fluentbit.loggingRef -}}
+  {{- $fluentbitSpec := set $fluentbitSpec "loggingRef" (default .Values.logging.loggingRef .Values.logging.fluentbit.loggingRef) -}}
```
Comment (aslafy-z) [inline charts/logging-operator/templates/logging_fluentbit.yaml:5]:
```suggestion
{{- if .Values.logging.loggingRef -}}
  {{- $fluentbitSpec := set $fluentbitSpec "loggingRef" (default .Values.logging.loggingRef .Values.logging.fluentbit.loggingRef) -}}
```

File: charts/logging-operator/templates/logging_fluentbit.yaml [hunk 5]
```diff
@@ -1,11 +1,17 @@
...
+{{- end -}}
+
 apiVersion: logging.banzaicloud.io/v1beta1
 kind: FluentbitAgent
 metadata:
   name: {{ include "logging-operator.releasename" . }}
   labels: {{ include "logging-operator.labels" . | nindent 4 }}
 {{- if .Values.logging.fluentbit }}
-spec: {{ toYaml .Values.logging.fluentbit | nindent 2 }}
+spec: {{- toYaml .Values.logging.fluentbit | nindent 2 }}
```
Comment (aslafy-z) [inline charts/logging-operator/templates/logging_fluentbit.yaml:14]:
```suggestion
spec: {{- toYaml $fluentbitSpec | nindent 2 }}
```

**Review by pepov — APPROVED** [review]
Nice!

**Pull request merged**